typed-racket-lib/typed-racket/static-contracts/combinators/structural.rkt

```diff
@@ -156,8 +157,12 @@
   ((listof/sc (#:covariant)) listof #:flat)
   ((cons/sc (#:covariant) (#:covariant)) cons/c #:flat)
   ((set/sc (#:covariant #:chaperone)) set/c #:flat)
-  ((vector/sc . (#:invariant)) vector/c #:chaperone)
+  ((immutable-vectorof/sc (#:covariant)) immutable-vectorof/c #:flat)
+  ((mutable-vectorof/sc (#:invariant)) mutable-vectorof/c #:chaperone)
   ((vectorof/sc (#:invariant)) vectorof #:chaperone)
+  ((immutable-vector/sc . (#:covariant)) immutable-vector/c #:chaperone) ;; immutable-vector/c is not flat, see `utils/vector-contract.rkt`
```

Comment:
but it could be flat, right? if all its elements are flat?

Reply:
(maybe there's not a good way to say 'maybe-flat' currently though...?

Reply:
Right, I didn't see a good way.